metaflow/plugins/aws/step_functions/schedule_decorator.py

```diff
@@ -1,22 +1,73 @@
 from metaflow.decorators import FlowDecorator
+from .event_bridge_client import EventBridgeClient
+from .step_functions_client import StepFunctionsClient
+class ScheduleDefinition():
 
+    def __init__(self, schedule_obj):
+        pass
+
+    def generate_eventbridge_rule(self):
+        pass
+
+class CronScheduleDefinition(ScheduleDefinition):
+
+    def __init__(self, schedule_obj):
+        self.cron = schedule_obj
+
+    def generate_eventbridge_rule(self, name, state_machine_arn, sfn_access_role):
+        return (
+            EventBridgeClient(name)
+                .cron(self.cron)
+                .role_arn(sfn_access_role)
+                .state_machine_arn(state_machine_arn)
+                .schedule()
+        )
+
+class FlowTriggerScheduleDefinition(ScheduleDefinition):
+
+    def __init__(self, schedule_obj):
+        self.flow_name = schedule_obj
+
+    def _get_state_machine_arn(self):
+
+        flow = StepFunctionsClient().search(self.flow_name)
+
+        if flow:
+            return flow.get('stateMachineArn', None)
+        else:
+            raise Exception("Dependant State Machine Found")
+
+    def generate_eventbridge_rule(self, name, state_machine_arn, sfn_access_role):
+
+        dependant_state_machine_arn = self._get_state_machine_arn()
+
+        return (
+            EventBridgeClient(name)
+                .dependent_state_machine_arn(dependant_state_machine_arn)
+                .role_arn(sfn_access_role)
+                .state_machine_arn(state_machine_arn)
+                .schedule()
+        )
 
 class ScheduleDecorator(FlowDecorator):
     name = "schedule"
-    defaults = {"cron": None, "weekly": False, "daily": True, "hourly": False}
+    defaults = {"cron": None, "weekly": False, "daily": True, "hourly": False, "flow_name": None}
```

Comment:
@iiAnderson - What's your opinion on instead of overriding `@schedule` - we introduce `@TriggerOnFinish`? We can make sure that `@TriggerOnFinish` obeys Metaflow namespace.

Reply:
@npow - this could be potentially interesting for you as well.

Reply:
@savingoyal An interesting idea, I've been playing around with how we could tie all of the Events into EventBridge - so you could essentially implement a `@TriggerOn`, then select an event to trigger on.

When a flow finishes/fails/succeeds you could then push events to Eventbridge, which then triggers dependent flows.

I think that makes it more of a complete framework for event handling, rather than just a single "Succeeded" as it currently is. I don't know how it plays into other Compute environments though (as it's certainly possible for Batch).

Reply:
Probably out of scope for this PR, but we were thinking about triggering arbitrary resources on Step Function completion (eg. Glue Crawlers, Lambda functions, etc). Any thoughts on how this would fit into the design?

Reply:
If you built the system using Eventbridge this would be easy, as they all tie into it. We would just need to implement a definition language to define these events within a decorator in Metaflow. Designing this to be user-friendly would be challenging though.